src/Frame.cpp

```diff
@@ -234,18 +500,81 @@ const std::unordered_map<std::string, double> BOND_GUESSING_RADII = {
 };
 
 optional<double> guess_bonds_radius(const Atom& atom) {
-    const auto& type = atom.type();
-    auto it = BOND_GUESSING_RADII.find(type);
-    if (it != BOND_GUESSING_RADII.end()) {
-        // allow configuration file to override data from BOND_GUESSING_RADII
-        auto user_config = Configuration::atom_data(type);
-        if (user_config && user_config->vdw_radius) {
-            return user_config->vdw_radius.value();
-        } else {
-            return it->second;
-        }
-    } else {
-        // default to chemfiles provided atom type
-        return atom.vdw_radius();
-    }
+   const auto& type = atom.type();
+   auto it = BOND_GUESSING_RADII.find(type);
+   if (it != BOND_GUESSING_RADII.end()) {
+      // allow configuration file to override data from BOND_GUESSING_RADII
+      auto user_config = Configuration::atom_data(type);
+      if (user_config && user_config->vdw_radius) {
+         return user_config->vdw_radius.value();
+      }
+      else {
+         return it->second;
+      }
+   }
+   else {
+      // default to chemfiles provided atom type
+      return atom.vdw_radius();
+   }
+}
+
+int Frame::generateBonds(std::vector<int>& result, int totalCells, std::vector<std::vector<int>>& cells,
```

Comment:
same as above, `std::vector<std::vector>` have bad CPU cache behavior. 

For `cells`, each sub-vector size is fixed at 27, right? Then it could be `std::vector<std::array<int, 27>>`.

Reply:
This function could also directly add bonds to the frame instead of creating an extra "result" vector

Reply:
> same as above, `std::vector<std::vector>` have bad CPU cache behavior.
> 
> For `cells`, each sub-vector size is fixed at 27, right? Then it could be `std::vector<std::array<int, 27>>`.

No, here we don't know the size of the sub-vector where atoms are stored. The `cells` data structure stores the atoms in cells depending on their position. Maybe the name of the variable is not clear enough but with documentation it will be clearer for sure. Anyway, I will change the `std::vector<std::vector>` data structure to have a better CPU cache behavior.